pallets/msa/src/lib.rs

```diff
@@ -594,15 +633,23 @@ pub mod pallet {
 		///
 		/// # Errors
 		/// * [`Error::NoKeyExists`] - origin does not have an MSA
-		/// * [`Error::ExceedsMaxProviderNameSize`] - Too long of a provider name
 		/// * [`Error::DuplicateProviderRegistryEntry`] - a ProviderRegistryEntry associated with the given MSA id already exists.
 		///
 		#[pallet::call_index(2)]
 		#[pallet::weight(T::WeightInfo::create_provider())]
-		pub fn create_provider(origin: OriginFor<T>, provider_name: Vec<u8>) -> DispatchResult {
+		pub fn create_provider(
```

Comment:
Did we make a decision to make these breaking changes, rather than do a `create_provider_v2`? This will, at minimum, break Provider Dashboard and many E2E tests across many repos...

Reply:
Good suggestion, This definitely a breaking change, thanks for putting in perspective. @wilwade  are you comfortable making this change all the way up.

My impression is to keep it in feature branch will all necessary changes arent not incorporated, since this only affects Provider Dashboard, and merge all together, if it was affecting gateway I would say yes go with v2 first

Reply:
On a second thought we should maintain both versions

1. Will still take name and put in registry 
2. Can accept a payload 
This could give good runway for changes to reach other dependencies

Reply:
so we can retrain old extrinsic and create a `_v2` for each and is just a re-routing? we can mark them `#[deprecated(note = "please use `verify` instead")]`

Reply:
That's the usual pattern when doing `_v2`, etc... unless there's a clear reason not to? Only Provider Dashboard is likely affected by `propose` and `..._via_governance`, but I'm not a fan of cutting corners on that kind of pattern. I'll defer to @wilwade .